### ISSUE TITLE
Incorrect Line Separator for Windows in text_formatter

### ISSUE THREAD

**Issue by ZurgInq** [sirupsen/logrus#637]
To separate strings, one character is used without checking the current OS

https://github.com/sirupsen/logrus/blob/v1.0.3/text_formatter.go#L123
```go
func (f *TextFormatter) Format(entry *Entry) ([]byte, error) {
...
b.WriteByte('\n') //but for Windows OS need "\r\n"
return b.Bytes(), nil
```

**Comment by maheshdotwaidande99** [comment]
@ZurgInq You are findings and observation are correct even I am also face a same issue when I try to open log file from notepad but I did not observe  same behavior with modern editors like notepad++ and vim/gvim. Logrus work well with modern editors like (Notepad++/gvim/vim) etc. 

I have made appropriate code changes to fix this issue. I wanted to upstream my code changes and here I required help from community members. I have couple of questions here, is there any standard procedure to upstream code change? OR   Is there is link which I need to follow to contribute my work. 

Steps followed by me till now,
i. Fork logrus repository 
ii. Create a new branch (logrus_637) from master 
Here is link of my code changes, https://github.com/maheshdotwaidande99/logrus/tree/logrus_637 . Before generating pull request I want to compliant with procedure and seeking guidance here.
